tests/accessor/accessor_common.h

```diff
@@ -0,0 +1,578 @@
+  return access_modes;
+}
+
+/**
+ * @brief Factory function for getting type_pack with dimensions values
+ */
+inline auto get_dimensions() {
+  static const auto dimensions = integer_pack<1, 2, 3>::generate_unnamed();
+  return dimensions;
+}
+
+/**
+ * @brief Factory function for getting type_pack with target values
+ */
+inline auto get_targets() {
+  // FIXME: re-enable when sycl::target::host_task is supported
+#if !defined(__HIPSYCL__) && !defined(__COMPUTECPP__) && \
+    !defined(__SYCL_COMPILER_VERSION)
+  static const auto targets =
+      value_pack<sycl::target, sycl::target::device,
+                 sycl::target::host_task>::generate_named("target::device",
+                                                          "target::host_task");
+#else
+  static const auto targets =
+      value_pack<sycl::target, sycl::target::device>::generate_named(
+          "target::device");
+#endif
+
+  return targets;
+}
+
+/**
+ * @brief Function helps to generate type_pack with sycl::vec of all supported
+ * sizes
+ */
+template <typename T, typename StrNameType>
+inline auto add_vectors_to_type_pack(StrNameType type_name) {
+  return named_type_pack<
+      T, sycl::vec<T, 1>, sycl::vec<T, 2>, sycl::vec<T, 4>, sycl::vec<T, 8>,
+      sycl::vec<T, 16>>::generate(type_name, "vec<" + type_name + ", 1>",
+                                  "vec<" + type_name + ", 2>",
+                                  "vec<" + type_name + ", 4>",
+                                  "vec<" + type_name + ", 8>",
+                                  "vec<" + type_name + ", 16>");
+}
+
+/**
+ * @brief Contains helper functions for change and compare generic types and
+ * sycl::vec
+ *
+ */
+namespace value_helper {
+/**
+ * @brief Returns result of comparing of left and right
+ * @return Returns true if left == right, false otherwise
+ */
+template <typename T, typename U = T>
+inline bool compare_vals(const T& left, const U& right) {
+  return left == right;
+}
+
+/**
+ * @brief Returns result of comparing of left and right
+ * @return Returns true if left == right, false otherwise
+ */
+template <typename T>
+inline bool compare_vals(const T& left, const T& right) {
+  return check_equal_values(left, right);
+}
+
+/**
+ * @brief Returns result of comparing of left and right
+ * @return Returns true if left == right, false otherwise
+ */
+template <typename T, typename U = T, int N>
+inline bool compare_vals(const sycl::vec<T, N>& left, const U& right) {
+  bool are_equal = true;
+  for (size_t i = 0; i < N; ++i) {
+    are_equal &= left[i] == right;
+  }
+  return are_equal;
+}
+
+/**
+ * @brief Returns result of comparing of left and right
+ * @return Returns true if left == right, false otherwise
+ */
+template <typename T, typename U = T, int N>
+inline bool compare_vals(const sycl::marray<T, N>& left, const U& right) {
+  bool are_equal = true;
+  for (size_t i = 0; i < N; ++i) {
+    are_equal &= left[i] == right;
+  }
+  return are_equal;
```

Comment:
Idem

Reply:
More generally I have the feeling that all these double definitions for `sycl::vec` and `sycl::marray` could be factorized.
Perhaps by adding the equivalent of a C++20 concept true for `sycl::vec` or `sycl::marray`, but implemented by `std::enable_if_t`.

Reply:
As it's yet another one implementation of `value_helper` functions ([first impl](https://github.com/KhronosGroup/SYCL-CTS/blob/b00bde0e3580d0c8a3cd0c7f7c46f60c5cb4d596/tests/extension/oneapi_device_global/device_global_common.h#L84) in `device_global` tests), I moved these functions in separate file in PR #324